### PR TITLE
Add new types

### PR DESCRIPTION
**Pull request by kostyanf14**
@muflone Interfaces tests will expectedly fail because `control-demo.kerio.com` is unavailable.

### CODE REVIEW

**Comment by kostyanf14** [comment]
Hi @muflone,

Can you please review this PR?

**Comment by muflone** [comment]
Hi @kostyanf14 
Thank you for your contribution

File: pykerio/enums/certificate_type.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License
+#  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+##
+
+from . import BaseEnumeration
```
Comment (muflone) [inline pykerio/enums/certificate_type.py]: please maintain the full imports path as
from pykerio.enums.base_enumeration import BaseEnumeration

File: pykerio/interfaces/reverse_proxy.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License
+#  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+##
+
+from ..pykerio import PyKerio
```
Comment (muflone) [inline pykerio/interfaces/reverse_proxy.py]: please be coherent with the explicit imports

File: pykerio/structs/reverse_proxy_rule.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License
+#  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+##
+
+from . import BaseStruct
```
Comment (muflone) [inline pykerio/structs/reverse_proxy_rule.py]: also here, please explicit imports

File: pykerio/structs/valid_period.py [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License
+#  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+##
+
+from . import BaseStruct
```
Comment (muflone) [inline pykerio/structs/valid_period.py]: explicit imports

**Review by muflone — COMMENTED** [review]
Please could you replace the imports with absolute imports?

**Comment by kostyanf14** [comment]
@muflone Fixed imports

File: pykerio/lists/reverse_proxy_rule_list.py [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+#  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+##
+
+from pykerio.lists.base_list import BaseList
+from pykerio.structs.reverse_proxy_rule import ReverseProxyRule
+
+
+class ReverseProxyRuleList(BaseList):
+    def __init__(self, *args, **kwargs):
+        BaseList.__init__(self, ReverseProxyRule, *args, **kwargs)
```
Comment (muflone) [inline pykerio/lists/reverse_proxy_rule_list.py:27]: this one could be simplified using

`super().__init__(ReverseProxyRule, *args, **kwargs)`

But it's a general change to apply in many parts, also for my code, in structs

I'll fix it after completing the merge

File: pykerio/structs/reverse_proxy_config.py [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License
+#  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+##
+
+from pykerio.structs.base_struct import BaseStruct
+
```
Comment (muflone) [inline pykerio/structs/reverse_proxy_config.py:22]: empty line unneeded
I'll fix it after the merge

File: pykerio/structs/reverse_proxy_config.py [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+from pykerio.lists.reverse_proxy_rule_list import ReverseProxyRuleList
+
```
Comment (muflone) [inline pykerio/structs/reverse_proxy_config.py:24]: empty line unneeded
I'll fix it after the merge

File: pykerio/structs/reverse_proxy_rule.py [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License
+#  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+##
+
+from pykerio.structs.base_struct import BaseStruct
+
```
Comment (muflone) [inline pykerio/structs/reverse_proxy_rule.py:22]: empty line unneeded
I'll fix it after the merge

File: pykerio/structs/reverse_proxy_rule.py [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+from pykerio.enums.https_server_mode import HttpsServerMode
+
```
Comment (muflone) [inline pykerio/structs/reverse_proxy_rule.py:24]: empty line unneeded
I'll fix it after the merge

File: pykerio/structs/reverse_proxy_rule.py [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+from pykerio.structs.id_reference import IdReference
+
```
Comment (muflone) [inline pykerio/structs/reverse_proxy_rule.py:26]: empty line unneeded
I'll fix it after the merge

File: pykerio/structs/reverse_proxy_rule.py [hunk 11]
```diff
@@ -0,0 +1,43 @@
...
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License
+#  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+##
+
+from pykerio.structs.base_struct import BaseStruct
```
Comment (muflone) [inline pykerio/structs/reverse_proxy_rule.py:21]: structs go after the enums
I'll fix it after the merge

File: pykerio/structs/valid_period.py [hunk 12]
```diff
@@ -0,0 +1,37 @@
...
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License
+#  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+##
+
+from pykerio.structs.base_struct import BaseStruct
```
Comment (muflone) [inline pykerio/structs/valid_period.py:21]: imports order
I'll fix it after the merge

**Review by muflone — APPROVED** [review]
it seems ok

**Comment by muflone** [comment]
thank you again, merged

File: pykerio/structs/valid_period.py (hunk 12 above)
Comment (kostyanf14) [reply pykerio/structs/valid_period.py:21]: @muflone 

I suggest you add `isort` to the CI check. This tool validate and fix import order. Very simple usage for GitHub https://pycqa.github.io/isort/docs/configuration/github_action.html

**Pull request merged**